addons/event_crm/models/event_crm.py

```diff
@@ -0,0 +1,104 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import fields, models
+from odoo.tools.safe_eval import safe_eval
+
+
+class EventCrmRule(models.Model):
+    """
+    When registrations are created, if they match rules, we create leads depending of the
+    lead_creation_basis and we link those leads to the registrations. The lead creation type
+    determines the number of leads to create: one per order or one per attendee.
+    A rule can have 4 types of restrictions:
+        - The company of the event link to the registration
+        - The type of the event
+        - The event itself
+        - A domain set on event.registration
+
+    When setting the rule you can choose multiple event types and/or one particular event, so
+    it's totally possible to have a rule on multiple event types and specified one event which
+    does not belong to this categories.
+    """
+    _name = "event.crm.rule"
+    _description = "Event CRM Rules"
+    _order = "sequence, id"
+
+    # Rule configuration fields
+    name = fields.Char('Rule Name', required=True, translate=True)
+    sequence = fields.Integer(default=10)
+    active = fields.Boolean('Active', default=True)
+    lead_creation_basis = fields.Selection([
+        ('attendee', 'Per Attendee'), ('order', 'Per Order')],
+        string='Create', default='attendee', required=True,
+        help='One per Attendee : A Lead is created for each Attendee (B2C).\n'
+            'One per Order : A single Lead is created per Ticket Batch/Sale Order (B2B)')
+    event_type_ids = fields.Many2many('event.type',
+        string='Event Categories',
+        help='Filter the attendees to include those of this specific event category. If not set, no event category restriction will be applied.')
+    event_id = fields.Many2one('event.event',
+        string='Event', domain="[('company_id', 'in', [company_id or current_company_id, False])]",
+        help='Filter the attendees to include those of this specific event. If not set, no event restriction will be applied.')
+    company_id = fields.Many2one('res.company', string='Company',
+        help="Restrict the trigger of this rule to events belonging to a specific company.\nIf not set, no company restriction will be applied.")
+    registration_filter_domain = fields.Text(string="Registrations Domain", default='[]', required=True,
```

Comment:
event_registration_filter ? That way all filtering fields begin with event_, except company that is a specific keyword.